### DISCUSSION TITLE
Jellyfin and multiple Artists ( semicolon? )

### DISCUSSION THREAD

**Discussion by Emporea** [beetbox/beets#4620] [Q&A]
I have tagged my music with the spotify plugin using --group-album option.
If an album or song has multiple artists the artist folder name and the artist tag use a comma ',' for seperation.
AFAIK jellyfin uses semicolons to recognize multiple artists.

Now I have multiple artists of the same artist, cause jellyfin thinks an artist with a comma and a second artist is another artist.

Does somebody know if I can change something in jellyfin?
Or is there a beets command that will write all tags with ";" instead of ","?
Will this help, since the folders would still contain "," as i think that its not allowed for a semicolon to be in a filepath?

**Comment by JOJ0** [comment]
You could try the `ftintitle` plugin. For example it translates ` artist1, artist2 - tracktitle1` to `artist1 - tracktitle1 (feat. artist2)`

I know it works with MusicBrainz-tagged files as well as Discogs tagged ones. Not sure if spotify-tagged will work. Just give it a try or ask @arsaboo whether they know by heart if it works :-)

All the best and HTH

**Comment by Emporea** [comment]
*hogwards meme*: why is it that if i ask a question its always you.

**Reply by JOJ0** [reply]
Did the ftintitle plugin help?

**Reply by Emporea** [reply]
I haven't tried this yet. For the beginning I am searching for a way to sperate artists via a semicolon inside the tags. Is there a way?

**Comment by Emporea** [comment]
I already have imported all my music.

I dont know if I failed to explain it or If I was unable to understand how ftintitle could help me.

If a song has multilpe artists they are seperated by a comma

```
Album                           : Delirium
Album Artist                    : Fatoni, Edgar Wasser
Albumartist                     : Fatoni, Edgar Wasser
Albumartist Credit              : 
Albumartistsort                 : 
Musicbrainz Albumcomment        : 
Releasestatus                   : official
Musicbrainz Albumstatus         : official
Releasetype                     : album
Musicbrainz Albumtype           : album
Arranger                        : 
Artist                          : Fatoni, Edgar Wasser
```

Jellyfin has problems recognizing comma in artist fields. ASAIK it need semicolons. Is there a beets configuration that lets me choose the artist seperator?

**Reply by JOJ0** [reply]
Separation of Artists is done by the metadata source plugin you are using. Some can do strings instead of simply comma as separaters, something like "feat" or "&" or "and".

This would result in artist tags like "artist1 feat artist2" for example.

And those can be translated by the ftintitle plugin to "artist1" in artist tag and "title feat artist2" in title tag.

Thus you'd get artist tags without comma

Unfortunately my first suggestion a couple of posts above https://github.com/beetbox/beets/discussions/4620#discussioncomment-4599906 was missing a piece: You dont have strings like "feat" as a separator because you used Spotify plugin which doesnt support that. It always uses comma afaik (assumption alert! you'd need to check the spotify code)

This function provides the words used by ftintitle as separators https://github.com/beetbox/beets/blob/6c06a18f8b63fff424749bc945d81f2cdc9d315b/beets/plugins.py#L494-L505

My assumption that a simple comma is a feat token as well was wrong. I didnt check the code before I suggested it to you. Sorry.

**Comment by arsaboo** [comment]
I would also look at the [import-replace](https://github.com/edgars-supe/beets-importreplace) plugin. It allows regex substitution at the time of import.

**Comment by TheRedCyclops** [comment]
For anyone looking at this in the future now jellyfin supports custom delimiters in the library settings, so for this use case there's no need to change the commas to semicolons

**Reply by LicioLab** [reply]
there are artists that have commas in their name. how would you go about that?

**Reply by TheRedCyclops** [reply]
After some more iterations what I finally did was enabling the use of the ARTISTS tag, that one has properly delimited artists and works pretty well, after that I enabled custom delimiters with only ";" and used it to separate genres manually since those don't seem to have an equivalent tag supported
